pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -92,6 +96,34 @@ type UpdateStrategy struct {
 	*RegistryPoll `json:"registryPoll,omitempty"`
 }
 
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface and thus supports custom unmarshaling for an updateStrategy.
+// UnmarshalJSON checks to see if the provided polling interval is valid. If so, use it, if not set it to a default interval.
+// This custom unmarshaling is required in the case where a user provides an invalid string for the polling interval.
+// An invalid interval causes informer operations to fail at the cache level since the unmarshalerDecoder cannot populate the concrete type.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) error {
+	var updateStrategy map[string]map[string]string
+	if err := json.Unmarshal(data, &updateStrategy); err != nil {
+		return err
+	}
+	if updateStrategy == nil {
+		return nil
+	}
+
+	registryPoll := &RegistryPoll{}
+	if rp, ok := updateStrategy["registryPoll"]; ok {
+		if i, ok := rp["interval"]; ok {
+			duration, err := time.ParseDuration(i)
+			if err != nil {
+				registryPoll.Interval = &metav1.Duration{Duration: defaultPollingDuration}
```

Comment:
Why don't we just return a custom error here instead? The CatalogSource reconciler can then display the unmarshalling error on it's status condition.

Reply:
I think I'm also a fan of just bubbling up a custom error at this point (also suggested the same thing in https://github.com/operator-framework/api/pull/165#discussion_r733916173). We could also use a custom error, bubble that up to the call site, and at the call site set the default registry interval if we're still interested in making the explicit decision to default to an interval during parsing failures. I think I'm largely in the corner we should avoid defaulting to an interval during a parsing failure, and instead, bubbling that up to the CatalogSource's status field.

@exdx @njhale Any thoughts?